### ISSUE TITLE
Collector makes offscreen rendering output zero images

### ISSUE THREAD

**Issue by navigator8972** [aai-institute/tianshou#800]
- [x] I have marked all applicable categories:
    + [x] exception-raising bug
    + [ ] RL algorithm bug
    + [ ] documentation request (i.e. "X is missing from the documentation.")
    + [ ] new feature request
- [x] I have visited the [source website](https://github.com/thu-ml/tianshou/)
- [x] I have searched through the [issue tracker](https://github.com/thu-ml/tianshou/issues) for duplicates
- [x] I have mentioned version numbers, operating system and environment, where applicable:
  ```python
  import tianshou, gym, torch, numpy, sys
  print(tianshou.__version__, gym.__version__, torch.__version__, numpy.__version__, sys.version, sys.platform)
  ```
Versions: 0.4.11 0.26.2 1.13.1 1.23.5 3.9.15 | packaged by conda-forge | (main, Nov 22 2022, 08:48:25) 
[Clang 14.0.6 ] darwin

I am trying to log offscreen rendering during training iterations while found that rendered images are all zeros after the first record episode. I tracked down the issue a bit and found a minimum example below can reproduce this issue. It seems related to collector of tianshou as using gym/gymnasium itself (commenting out two collector lines) gives expected output.

```python

import gym

from gym.wrappers.record_video import RecordVideo
from tianshou.data.collector import Collector

def capture_video_on_episode(episode_id):
    return True

native_env = gym.make("Humanoid", render_mode="rgb_array")

env = RecordVideo(native_env, "./", capture_video_on_episode)
env.reset()

#comment out any one of the below line of code for non-zero offscreen rendering
test_collector = Collector(None, env) 

#comment out any one of the below line of code for non-zero offscreen rendering
native_env = gym.make("Humanoid", render_mode="rgb_array")

#comment out any one of the below line of code for non-zero offscreen rendering
env = RecordVideo(native_env, "./", capture_video_on_episode)

#test_controller1 = Collect(None, env) will work
test_collector = Collector(None, env)

env.reset()
for _ in range(100):
    env.step(env.action_space.sample())
print(env.video_recorder.recorded_frames[10][:10, :10, 0])
env.close()
```

Basically, running above code snippet will give all zero screen output, which looks strange to me because creating/destructing collector should not impact the underlying gym environment. Commenting out some lines could generate non-zero output again though. I guess some interaction between collector and env goes expectedly but I don't find anything particularly suspicious as Collector.__init__() just calls env.reset(). Any thoughts?

**Comment by Markus28** [comment]
Hey, could you try it with a non-mujoco environment (maybe do "CartPole-v1")? I am aware of some rendering problems with Mujoco. CartPole seems to be working on my system (haven't tried Mujoco yet). I will investigate it further.

**Comment by Markus28** [comment]
The problem definitely comes from messing with two Mujoco environments at the same time. We had similar problems here: https://github.com/openai/gym/issues/2989. I'm not completely sure why tianshou causes problems, but apparently there is something going wrong with the garbage collection of the environment (?). 

One fix would be to actually call `env.close()` (also works if you do it on the collector's vector environment) before you lose all references to the environment. In general, it is recommended to `close` environments before they are garbage collected, but most often it's not relevant. 

Although this is very weird behaviour, I currently don't see how tianshou could fix this, or if it is even worth fixing (because you should `close` anyways).

**Comment by navigator8972** [comment]
Hi, sorry for the late response. I tried pygame rendering and it was working as expected in both this example and the loop of trainer. So I guess it is indeed more likely to be an issue with mujoco itself. RecorderVideo can write videos by calling `start_video_recoder()` which is invoked in `step()` so I think we don't need `env.close()` in trainer.

I am trying to seek a workaround by locating which step in trainer causes some similar effects as destructing/re-initializing env. Any hints on the parts that I might miss or anyone managed to log offscreen rendering during training iterations before?

**Comment by navigator8972** [comment]
The issue should be caused by dependencies. Now can be solved with a workaround. See the linked issue of Gymnasium.
